core/src/main/java/org/elasticsearch/search/aggregations/AggregatorParsers.java

```diff
@@ -131,6 +131,7 @@ public AggregatorParsers(ParseFieldRegistry<Aggregator.Parser> aggregationParser
                     if (token == XContentParser.Token.VALUE_STRING || token == XContentParser.Token.VALUE_EMBEDDED_OBJECT) {
                         byte[] source = parser.binaryValue();
                         binaryParser = XContentFactory.xContent(source).createParser(source);
+                        binaryParser.setParseFieldMatcher(parseContext.parseFieldMatcher());
```

Comment:
seeing this I am wondering if this getter makes it confusing, shall we consider dropping it in favour of doing parseContext.parser().parseFieldMatcher() ? Or maybe it's not a big deal?